tensorflow/core/kernels/dml_resize_op.cc

```diff
@@ -127,23 +115,9 @@ class DmlResizeKernel : public DmlKernel {
     DataType tf_input_data_type = ctx->GetInputDataType(0);
     DML_TENSOR_DATA_TYPE dml_input_data_type = result.GetOutputDesc().dataType;
 
-    bool is_identity = input_tensor_shape == output_tensor_shape;
-
-    // ResizeBilinear only supports DT_FLOAT as an output type, and DML doesn't
-    // support mixed data types for Resample. Therefore, we need to cast all
-    // non-float input types to DML_TENSOR_DATA_TYPE_FLOAT32.
-    bool cast_to_float =
-        CastInputToFloat(interpolation_mode, tf_input_data_type, is_identity);
```

Comment:
Ah, so we're abandoning casting from uint8 to float for now? Yeah, DML doesn't support uint8 directly. (I wonder if transparently casting inside DirectML.dll would be useful, or if it would be better to keep always requiring explicit casting from the caller first. 🤔)

Reply:
I realized that CUDA doesn't register those types so we were unnecessarily burdening ourselves with unnecessary hacks.